### PR TITLE
RepositoryApi: Listing all repos regardless of project ( copy of #191) 

### LINKED ISSUE #190: RepositoryApi: get all public/private repos together without looping projects
RepositoryApi#list throws exception when sending project as null. I know that Bitbucket Rest API supports listing all repos together without looping through all projects one by one.

I couldn't find anything about that, so would like to know if there is a planned roadmap about this or if it already exists, and if the exception thrown by RepositoryApi#list is thrown intentionally or if it is a bug.

One more note, Bitbucket supports creating repositories without associating with a project.

### PR DESCRIPTION
**Pull request by mronus**
- Update base path for RepositoryApi
- Create a new endpoint in RepositoryApi to list all repositories regardless of project

closes #190

### CODE REVIEW

File: src/test/java/com/cdancy/bitbucket/rest/features/RepositoryApiLiveTest.java [hunk 1]
```diff
@@ -105,6 +105,25 @@ public void testListRepositories() {
...
+        boolean found = false;
+        for (final Repository possibleRepo : repositoryPage.values()) {
+            if (possibleRepo.name().equals(repoKey)) {
+                found = true;
+                break;
+            }
+        }
+        assertThat(found).isTrue();
+    }
+
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/RepositoryApiLiveTest.java]: Lets add a live test to exercise the failure case. I guess you _could_ put in say bogus values for things just to get bitbucket to return an error state. I do this for every endpoint so there should be multiple examples within this file.

File: src/test/java/com/cdancy/bitbucket/rest/features/RepositoryApiMockTest.java [hunk 2]
```diff
@@ -325,6 +325,68 @@ public void testGetRepositoryListNonExistent() throws Exception {
...
+            assertThat(repositoryPage.limit()).isEqualTo(limit);
+            assertThat(repositoryPage.isLastPage()).isFalse();
+            assertThat(repositoryPage.nextPageStart()).isEqualTo(size);
+            assertThat(repositoryPage.values()).hasSize(size);
+            assertThat(repositoryPage.values()).hasOnlyElementsOfType(Repository.class);
+        } finally {
+            server.shutdown();
+        }
+    }
+
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/RepositoryApiMockTest.java]: Same thing here: lets add a live test to exercise the failure case. I guess you _could_ put in say bogus values for things just to get bitbucket to return an error state. I do this for every endpoint so there should be multiple examples within this file.

File: src/test/java/com/cdancy/bitbucket/rest/features/RepositoryApiLiveTest.java (hunk 1 above)
Comment (mronus) [reply src/test/java/com/cdancy/bitbucket/rest/features/RepositoryApiLiveTest.java]: Done. Please review the last commit when you have time.

File: src/test/java/com/cdancy/bitbucket/rest/features/RepositoryApiMockTest.java (hunk 2 above)
Comment (mronus) [reply src/test/java/com/cdancy/bitbucket/rest/features/RepositoryApiMockTest.java]: Done. Please review the last commit when you have time.

**Comment by cdancy** [comment]
Thanks for the PR @mronus . Kicking a point release shortly.

**Comment by cdancy** [comment]
`v2.5.3` has been released with these changes.

**Pull request merged**